charmcraft/config.py

```diff
@@ -150,7 +171,18 @@ def format_pydantic_errors(errors):
         formatted_loc = format_pydantic_error_location(error["loc"])
         formatted_msg = format_pydantic_error_message(error["msg"])
 
-        combined.append(f"- {formatted_msg} in field {formatted_loc!r}")
+        if formatted_msg == "field required":
+            field_name, location = printable_field_location_split(formatted_loc)
+            combined.append(
+                f"- field {field_name} required in {location} configuration"
+            )
+        elif formatted_msg == "extra fields not permitted":
+            field_name, location = printable_field_location_split(formatted_loc)
+            combined.append(
+                f"- extra field {field_name} not permitted in {location} configuration"
+            )
+        else:
+            combined.append(f"- {formatted_msg} in field {formatted_loc!r}")
```

Comment:
This kind of handling would be useful in craft-parts too. Perhaps we could move it to an external common toolbox later.

Reply:
Sounds like a good idea